idaes/apps/grid_integration/model_data.py

```diff
@@ -0,0 +1,348 @@
+
+        """
+        Property getter for generator's name.
+
+        Returns:
+            str: generator's name
+        """
+
+        return self._gen_name
+
+    @gen_name.setter
+    def gen_name(self, value):
+
+        """
+        Property setter for generator's name.
+
+        Args:
+            value: generator's name in str
+
+        Returns:
+            None
+        """
+
+        if not isinstance(value, str):
+            raise TypeError(
+                f"Value for generator names must be str, but {type(value)} is provided."
+            )
+        self._gen_name = value
+
+    @property
+    def generator_type(self):
+
+        """
+        Property getter for generator's type.
+
+        Returns:
+            str: generator's type
+        """
+
+        return self._generator_type
+
+    @generator_type.setter
+    def generator_type(self, value):
+
+        """
+        Property setter for generator's type.
+
+        Args:
+            value: generator's type in str
+
+        Returns:
+            None
+        """
+
+        allowed_types = ["thermal", "renewable"]
+        if value not in allowed_types:
+            raise ValueError(
+                f"Value for generator types must be one of {allowed_types}, but {value} is provided."
+            )
+        self._generator_type = value
+
+    def _check_empty_and_sort_cost_pairs(self, pair_description, pairs):
+
+        """
+        Check whether a list of pairs for production and startup costs is empty,
+        and then sort them based on power output and startup lag.
+
+        Args:
+            pair_description: a string description for the list of pairs
+            pairs: a list of pairs
+
+        Returns:
+            None
+
+        """
+
+        if pairs is None or len(pairs) == 0:
+            raise ValueError(f"Empty {pair_description} are provided.")
+
+        # sort based on power output
+        pairs.sort(key=lambda p: p[0])
+
+        return
+
+    def _assemble_default_cost_bids(self, production_cost_bid_pairs):
+
+        """
+        Assemble the default production cost bids.
+
+        Args:
+            production_cost_bid_pairs: a list of pairs, which consit of power output
+            and corresponding marginal costs
+
+        Returns:
+            dict: the assembled production cost bids
+        """
+
+        self._check_empty_and_sort_cost_pairs(
+            pair_description="production cost pairs", pairs=production_cost_bid_pairs
+        )
+
+        if production_cost_bid_pairs[0][0] != self.p_min:
+            raise ValueError(
+                f"The first power output in the bid should be the Pmin {self.p_min}, but {production_cost_bid_pairs[0][0]} is provided."
+            )
+
+        if production_cost_bid_pairs[-1][0] != self.p_max:
+            raise ValueError(
+                f"The last power output in the bid should be the Pmax {self.p_max}, but {production_cost_bid_pairs[-1][0]} is provided."
+            )
```

Comment:
I think its fine to do this validation, but its no longer required for Egret/Prescient.